src/main/java/org/craftercms/studio/impl/v1/service/security/AuthenticationHeadersSecurityProvider.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Crafter Studio Web-content authoring solution
+ * Copyright (C) 2007-2017 Crafter Software Corporation.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.craftercms.studio.impl.v1.service.security;
+
+import org.apache.commons.lang3.StringUtils;
+import org.craftercms.commons.http.RequestContext;
+import org.craftercms.studio.api.v1.constant.DmConstants;
+import org.craftercms.studio.api.v1.constant.StudioConstants;
+import org.craftercms.studio.api.v1.dal.Group;
+import org.craftercms.studio.api.v1.dal.SiteFeed;
+import org.craftercms.studio.api.v1.dal.User;
+import org.craftercms.studio.api.v1.exception.SiteNotFoundException;
+import org.craftercms.studio.api.v1.exception.security.*;
+import org.craftercms.studio.api.v1.log.Logger;
+import org.craftercms.studio.api.v1.log.LoggerFactory;
+import org.craftercms.studio.api.v1.service.activity.ActivityService;
+
+import javax.servlet.http.HttpServletRequest;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+import static org.craftercms.studio.api.v1.util.StudioConfiguration.*;
+
+public class AuthenticationHeadersSecurityProvider extends DbWithLdapExtensionSecurityProvider {
+
+    private final static Logger logger = LoggerFactory.getLogger(AuthenticationHeadersSecurityProvider.class);
+
+    @Override
+    public String authenticate(String username, String password) throws BadCredentialsException, AuthenticationSystemException {
+        if (isAuthenticationHeadersEnabled()) {
+            RequestContext requestContext = RequestContext.getCurrent();
+            if (requestContext != null) {
```

Comment:
Please add a debug log to indicate that we're indeed about to authenticate via headers.